### PR TITLE
Documentation Update: Capture `cloud`/ `backend` block override behavior in docs

### PR DESCRIPTION
**Pull request by GoodmanBen**
**Current status**:
The code that implements the behavior needing additional documentation can be found [here](https://github.com/hashicorp/terraform/blob/main/internal/configs/module.go#L379).

The behavior is as follows:
When merging in a `cloud` backend block within the `terraform` block from an `override.tf ` file, the backend used is from that `cloud` block, regardless of whether the original `terraform` block had a `backend` or `cloud` block specified. Similarly, when mergining in a `backend` backend block within the `terraform` block from an `override.tf` file, the backend used is from that `backend` block, regardless of whether the original `terraform` block had a `cloud` or `backend` backend block specified.

The observed behavior is certainly a nice feature, it just is a unique merging pattern and could use better documentation.
Current documentation does not mention this behavior. In fact, there is no indication in the merging [docs](https://www.terraform.io/language/files/override#merging-terraform-blocks) or Terraform Cloud [docs](https://www.terraform.io/language/settings/terraform-cloud) that this behavior should be expected. 

Missing complete documentation on this behavior led to difficulties while looking through the documentation mentioned as captured [here](https://github.com/minamijoyo/tfmigrate/issues/75#issuecomment-1064134043) and realizing that it did not fully capture Terraform's merging behavior when backend overriding.

**This PR**:
Add the additional documentation needed to make clear the expected behavior when overriding a `backend` block with a `cloud` block and vice a versa.

**Testing**:
The only changes in this PR are to markdown files. All prior tests passing.

### CODE REVIEW

File: website/docs/language/settings/terraform-cloud.mdx
Comment (laurapacilio) [inline website/docs/language/settings/terraform-cloud.mdx:33]:
```suggestion
A configuration can only provide one `cloud` block, and the `cloud` block cannot refer to named values like input variables, locals, or data source attributes. However, if a state backend is defined in an [override file](/language/files/override), the state backend will override an existing `cloud` block in the configuration upon merging.
```

File: website/docs/language/files/override.mdx
Comment (laurapacilio) [inline website/docs/language/files/override.mdx:157]:
```suggestion
The presence of a `backend` block in either the original configuration or an override file always takes precedence over a [Terraform CLI integration `cloud` block](/docs/language/settings/terraform-cloud). If a `cloud` block is set within the original configuration and a `backend` block is set in the override file, Terraform will use the backend specified in the override file upon merging. Similarly, if a `backend` block is set within the original configuration and a `cloud` block is set in the override file, Terraform will use the backend specified in the original configuration upon merging. 
```

**Review by laurapacilio — CHANGES_REQUESTED** [review]
Hi! Thank you so much for submitting this! I'm suggesting a couple of edits to hopefully make this even clearer and added a link of the override page so folks can find out more about the `cloud` block if they're not familiar.  Let me know what you think!

I'm also going to ask someone from terraform cloud engineering to provide a review.

File: website/docs/language/settings/terraform-cloud.mdx
Comment (GoodmanBen) [reply website/docs/language/settings/terraform-cloud.mdx:33]: 👍

File: website/docs/language/files/override.mdx
Comment (GoodmanBen) [reply website/docs/language/files/override.mdx:157]: I like the additional sentence at the beginning, but it raised to me an error in the original documentation suggestion that I wrote. I've fixed that error, while incorporating your suggestion into the beginning.

Comment (GoodmanBen) [reply website/docs/language/files/override.mdx]: Also kept the write-up changes to occur over multiple lines, in order to match toughly the line lengths in the rest of the documentation.

File: website/docs/language/settings/terraform-cloud.mdx
Comment (nfagerlund) [inline website/docs/language/settings/terraform-cloud.mdx:34]: I don't think this page actually needs this information -- as far as I know, we don't interject with details about `_override.tf` file block merging for any other block types, and I think it makes sense to keep that confined to just the override files page.

**Review by nfagerlund — COMMENTED** [review]
This looks accurate, now that you've fixed the copy-paste glitch! In general we've been trying not to refer to the `cloud` integration as a "backend," but this override behavior is one of only a couple spots where the shared code behaviors between cloud settings and backends are too entangled to really separate, so I'm fine with the wording.

My only request for changes is to not add info about override files to other pages, like the cloud settings page -- overriding is weird off-road behavior that you shouldn't have to think about very often, and this cloud/backend full-block-replacement thing is like... take that but triple it. If you need to know about that, you'll have gone to the override files page anyway.

I'll leave this to @laurapacilio for final approval call!

File: website/docs/language/settings/terraform-cloud.mdx
Comment (GoodmanBen) [reply website/docs/language/settings/terraform-cloud.mdx]: Yeah that makes a lot sense, have made this change.

**Review by laurapacilio — APPROVED** [review]
Thank you!

**Pull request merged**
